tests/platform_helper/utils/test_validation.py

```diff
@@ -419,6 +423,208 @@ def test_validate_s3_bucket_name_multiple_failures():
 
 
 @patch("dbt_platform_helper.utils.validation.warn_on_s3_bucket_name_availability", new=Mock())
-def test_validate_success(valid_platform_config):
-    PLATFORM_CONFIG_SCHEMA.validate(valid_platform_config)
+@patch("dbt_platform_helper.utils.validation.get_aws_session_or_abort", new=Mock())
+def test_validate_platform_config_success(valid_platform_config):
+    validate_platform_config(valid_platform_config)
     # No assertions - validate will error if config is invalid.
+
+
+@pytest.mark.parametrize(
+    "account, envs, exp_bad_envs",
+    [
+        ("non-prod", ["dev"], ["dev"]),
+        ("non-prod-acc", ["prod"], ["prod"]),
+        ("prod-acc", ["dev", "prod"], ["dev"]),
+        ("prod-acc", ["dev", "staging", "prod"], ["dev", "staging"]),
+        ("non-prod-acc", ["dev", "prod"], ["prod"]),
```

Comment:
I'm not convinced two of these scenarios add any value...
```suggestion
        ("prod-acc", ["dev", "staging", "prod"], ["dev", "staging"]),
        ("non-prod-acc", ["dev", "prod"], ["prod"]),
```